### PR TITLE
Add compute migration guide

### PR DESCRIPTION
**Pull request by kobajagi**
This PR adds a migration guide for `exoscale_compute`, `exoscale_network` and `exoscale_nic` resources to `exoscale_compute_instance` and `exoscale_private_network` resources.

Solves #277

### CODE REVIEW

File: docs/guides/migration-of-compute.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+We want to migrate `exoscale_compute` and `exoscale_nic` to the `exoscale_compute_instance` resource and `exoscale_network` to the `exoscale_private_network` resource.
+
+
+## Migration plan
+
+To achieve this migration, we have to remove all 3 deprecated resources from the state and then import them as `exoscale_compute_instance` and `exoscale_private_network`.
+
+## Applying the migration plan
+
+### Removing the security group from the state
```
Comment (jessicatoscani) [inline docs/guides/migration-of-compute.md]:
```suggestion
### Removing the resources from the state 
```

File: docs/guides/migration-of-compute.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+### Removing the security group from the state 
+
+This step is pretty straightforward: we will have to issue a `terraform state rm` command for each resources we have to remove from the state. In our example, we have to remove `exoscale_compute.my_instance`, `exoscale_network.my_network` and `exoscale_nic.my_nic`:
+
+```bash
+$ terraform state rm exoscale_compute.my_instance
+Removed exoscale_compute.my_instance
+Successfully removed 1 resource instance(s).
+
+q$ terraform state rm exoscale_network.my_network
```
Comment (jessicatoscani) [inline docs/guides/migration-of-compute.md]:
```suggestion
$ terraform state rm exoscale_network.my_network
```

File: docs/guides/migration-of-compute.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+## Migration plan
+
+To achieve this migration, we have to remove all 3 deprecated resources from the state and then import them as `exoscale_compute_instance` and `exoscale_private_network`.
+
+## Applying the migration plan
+
+### Removing the security group from the state 
+
+This step is pretty straightforward: we will have to issue a `terraform state rm` command for each resources we have to remove from the state. In our example, we have to remove `exoscale_compute.my_instance`, `exoscale_network.my_network` and `exoscale_nic.my_nic`:
```
Comment (jessicatoscani) [inline docs/guides/migration-of-compute.md]:
```suggestion
This step is pretty straightforward: we will have to issue a `terraform state rm` command for each resource we have to remove from the state. In our example, we have to remove `exoscale_compute.my_instance`, `exoscale_network.my_network` and `exoscale_nic.my_nic`:
```

File: docs/guides/migration-of-compute.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+resource "exoscale_private_network" "my_network" {
+  name = "privnet"
+  description = "Private Network"
+  zone = "ch-gva-2"
+}
+```
+
+In this example we are using unmanaged private network, for managed network you can copy values from `exoscale_network` definition with the following changes:
+- `display_text` must be renamed to `description`;
+- `network_offering` and `tags` are deprecated and should be removed.
```
Comment (jessicatoscani) [inline docs/guides/migration-of-compute.md]: `labels` exist in public-api to replace `tags`, we should implement them for the `exoscale_private_network` resource

File: docs/guides/migration-of-compute.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+│ Labels               │ n/a                                  │
+│ Reverse DNS          │                                      │
+┼──────────────────────┼──────────────────────────────────────┼
+```
+
+Instance ID is `0e28952c-1ae9-45fc-9f6d-0a9710175fd8` and we can import it:
+
+```bash
+$ terraform import exoscale_compute_instance.my_instance 0e28952c-1ae9-45fc-9f6d-0a9710175fd8@ch-gva-2
+
```
Comment (greut) [inline docs/guides/migration-of-compute.md:200]: What about taking advantage of the new features from 1.5?

https://developer.hashicorp.com/terraform/language/import

Comment (kobajagi) [reply docs/guides/migration-of-compute.md:200]: Thanks for suggestion @greut . I like the import blocks and imo it should be a recommended way to perform imports in the future. But I'd stick with classic import for this particular guide to be inline with the security group migration guide and give a bit more time to everyone to upgrade to 1.5+ version of terraform.

File: docs/guides/migration-of-compute.md (hunk 4 above)
Comment (kobajagi) [reply docs/guides/migration-of-compute.md]: Fixed

File: templates/guides/migration-of-compute.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+### Removing the security group from the state 
+
+This step is pretty straightforward: we will have to issue a `terraform state rm` command for each resources we have to remove from the state. In our example, we have to remove `exoscale_compute.my_instance`, `exoscale_network.my_network` and `exoscale_nic.my_nic`:
+
+```bash
+$ terraform state rm exoscale_compute.my_instance
+Removed exoscale_compute.my_instance
+Successfully removed 1 resource instance(s).
+
+q$ terraform state rm exoscale_network.my_network
```
Comment (sauterp) [inline templates/guides/migration-of-compute.md]:
```suggestion
$ terraform state rm exoscale_network.my_network
```

File: templates/guides/migration-of-compute.md [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+In this example we are using unmanaged private network, for managed network you can copy values from `exoscale_network` definition with the following changes:
+- `display_text` must be renamed to `description`;
+- `tags` must be renamed to `labels`;
+- `network_offering` is deprecated and should be removed.
+
+Now replace `exoscale_compute` and `exoscale_nic` blocks with `exoscale_compute_instance`:
+
+```hcl
+resource "exoscale_compute_instance" "my_instance" {
+	disk_size = 10
```
Comment (sauterp) [inline templates/guides/migration-of-compute.md]:
```suggestion
  disk_size = 10
```

**Review by sauterp — APPROVED** [review]
very clear!

**Review by jessicatoscani — APPROVED** [review]

File: docs/guides/migration-of-compute.md [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+- `display_text` must be renamed to `description`;
+- `tags` must be renamed to `labels`;
+- `network_offering` is deprecated and should be removed.
+
+Now replace `exoscale_compute` and `exoscale_nic` blocks with `exoscale_compute_instance`:
+
+```hcl
+resource "exoscale_compute_instance" "my_instance" {
+	disk_size = 10
+  name = "my-instance"
```
Comment (PhilippeChepy) [inline docs/guides/migration-of-compute.md]:
```suggestion
resource "exoscale_compute_instance" "my_instance" {
  disk_size = 10
  name = "my-instance"
```

File: templates/guides/migration-of-compute.md [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+  name = "my-instance"
```
Comment (PhilippeChepy) [inline templates/guides/migration-of-compute.md]:
```suggestion
  disk_size = 10
  name = "my-instance"
```

**Review by PhilippeChepy — APPROVED** [review]

**Pull request merged**